analyzers/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -35,7 +35,8 @@ public static bool IsMainMethod(this IMethodSymbol methodSymbol)
             // Adding support for new async main: https://blogs.msdn.microsoft.com/mazhou/2017/05/30/c-7-series-part-2-async-main
             return methodSymbol != null
                 && methodSymbol.IsStatic
-                && methodSymbol.Name.Equals("Main", StringComparison.OrdinalIgnoreCase) // VB.NET is case insensitive
+                && (methodSymbol.Name.Equals("Main", StringComparison.OrdinalIgnoreCase) // VB.NET is case insensitive
+                    || methodSymbol.Name.Equals("<Main>$", StringComparison.OrdinalIgnoreCase))
```

Comment:
Shouldn't we use here `StringComparison.Ordinal`? I think that top level method name is always `<Main>$` for C#.